urlchecker/client/__init__.py

```diff
@@ -141,6 +141,60 @@ def get_parser():
         default=5,
     )
 
+    # main check_ci entrypoint
+
+    check_ci = subparsers.add_parser(
```

Comment:
What does check_ci stand for? I can try to help to think of a name that better describes the function. If you are just checking a specific set of files, why not just add an argument to the parser for checking that takes files instead of patterns, etc.? The subparser here is a bit redundant in defining the same variables as check.

Reply:
`check_ci` was a total placeholder. Continuous Integration.

The initial reasoning for a separate command was that it would offer fewer options and be more straightforward to use (since specifying files should ignore file type checks and exclusions). Adding an option to check makes check feel pretty busy option-wise.

```bash
urlchecker check_specific README.md repositories.json
# Not ideal due to duplication, but usage is obvious.

urlchecker check --specific README.md repositories.json
# Awkward... it's basically --files but space-separated.
# What happens if the user specifies exclude patterns?
# - They're not used currently. Should the program exit non-zero?
```

Reply:
Why can't you just do `urlchecker check --files README.md,respositories.json` ?

Reply:
pre-commit doesn't provide the files to check in comma-separated format (only space-separated).

https://pre-commit.com/index.html#creating-new-hooks
- see the entry parameter. files are appended to the end of the string.

Reply:
So it looks like this files parameter is when to run the pre-commit hook, which is different than the files you'd want to run with urlchecker. So why does the urlchecker command need to have the same input and format as pre-commit?

Reply:
I've killed off the files parameter in a recent revision. I think I got confused about if the parser could handle lists of files vs patterns. You are correct that it wasn't needed.

pre-commit passes the list of git staged files to the hook (unless you run with -a or --all and then it checks all files that the hook matches).

pre-commit doesn't offer any way to transform the list of files to operate on... so we have to match it's expected format to an extent (a list of space separated files to check).

Reply:
My understanding from reading is that the files parameter for pre-commit hook indicates “run this custom script whenever one of these files changes.” And that is akin to a fit hook. So it would be logical to run it whenever any file changes to check an update of the site, or a specific folder pattern. I don’t see why you cannot use this pre-commit hook without making any changes to urlchecker as it is, using the extensive options that it offers.

Reply:
> I don’t see why you cannot use this pre-commit hook without making any changes to urlchecker as it is, using the extensive options that it offers.

It's due to the comma-separated --files argument in the current check subcommand. There's no way to get pre-commit to pass a comma-separated list.

Reply:
Additionally, the 'files' parameter in the config is not shared with the hook's executable (via env vars or otherwise). Files passed to the hook are the subset staged files that are in the 'files' parameter (or other pre-commit selection methods like type).